modules/api/test/LorisApiCandidatesTest.php

```diff
@@ -0,0 +1,307 @@
+            'string'
+        );
+        $this->assertSame(
+            gettype($candidatesCandidArray['Visits']),
+            'array'
+        );
+        $this->assertSame(
+            gettype($candidatesCandidArray['Visits']['0']),
+            'string'
+        );
+
+    }
+
+    /**
+     * Tests the HTTP POST request for the endpoint /candidates/{candid}
+     *
+     * @return void
+     */
+    public function testPostCandidatesCandid(): void
+    {
+        // First, create a valid new candidate
+        $json_new     = [
+            'Candidate' =>
+                [
+                    'Project' => "Rye",
+                    'Site'    => "Data Coordinating Center",
+                    'EDC'     => "2020-01-03",
+                    'DoB'     => "2020-01-03",
+                    'Sex'     => "Male"
+                ]
+        ];
+        $response_new = $this->client->request(
+            'POST',
+            "candidates",
+            [
+                'headers' => $this->headers,
+                'json'    => $json_new
+            ]
+        );
+        // Verify the status code
+        $this->assertEquals(201, $response_new->getStatusCode());
+        // Verify the endpoint has a body
+        $body = $response_new->getBody();
+        $this->assertNotEmpty($body);
+
+        // Second, create a candidate that already exist
+        $json_exist     = [
+            'Candidate' =>
+                [
+                    'Project' => "Pumpernickel",
+                    'Site'    => "Data Coordinating Center",
+                    'EDC'     => "2020-01-01",
+                    'DoB'     => "2020-01-01",
+                    'Sex'     => "Female"
+                ]
+        ];
+        $response_exist = $this->client->request(
+            'POST',
+            "candidates",
+            [
+                'headers' => $this->headers,
+                'json'    => $json_exist
+            ]
+        );
+        // Verify the status code
+        $this->assertEquals(201, $response_exist->getStatusCode());
+        // Verify the endpoint has a body
+        $body = $response_exist->getBody();
+        $this->assertNotEmpty($body);
+
+        // Finally, try to create a new candidate with an invalid input
+        $json_invalid = [
+            'Candidate' =>
+                [
+                    'Site' => "Data Coordinating Center",
+                    'EDC'  => "2020-01-03",
+                    'DoB'  => "2020-01-03",
+                    'Sex'  => "Male"
+                ]
+        ];
+
+        try{
+            $response_invalid = $this->client->request(
+                'POST',
+                "candidates",
+                [
+                    'headers' => $this->headers,
+                    'json'    => $json_invalid
+                ]
+            );
+            // Verify the status code
+            $this->assertEquals(201, $response_invalid->getStatusCode());
```

Comment:
I think the expected response status code is 400 Bad Request.

I think we could avoid the Exception throwing to simply look at the error.
```
$res = $client->request('GET', '/status/500', ['http_errors' => false]);
echo $res->getStatusCode();
```
see: http://docs.guzzlephp.org/en/stable/request-options.html#http-errors

Reply:
that's right, it's not 201. I'll try this way